coordinator/src/main/scala/filodb.coordinator/NamespaceCardinalityPublisher.scala

```diff
@@ -0,0 +1,103 @@
+package filodb.coordinator
+
+import java.util.concurrent.TimeUnit
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Success
+
+import akka.actor.ActorRef
+import kamon.Kamon
+import kamon.tag.TagSet
+import monix.execution.Scheduler.{global => scheduler}
+
+import filodb.coordinator.client.Client
+import filodb.coordinator.client.QueryCommands.LogicalPlan2Query
+import filodb.core.DatasetRef
+import filodb.query.{QueryResult, TsCardinalities}
+
+/**
+ * Periodically queries a node for all namespace cardinalities.
+ * Kamon gauges are updated with the response data.
+ *
+ * The intent is to publish a low-cardinality metric such that namespace
+ * cardinality queries can be efficiently answered.
+ */
+case class NamespaceCardinalityPublisher(dsIterProducer: () => Iterator[DatasetRef],
+                                         coordProducer: () => ActorRef) {
+  // All "_TU" constants quantify units of TIME_UNIT.
+  private val TIME_UNIT = TimeUnit.SECONDS
+  private val SCHED_INIT_DELAY_TU = 10
+  private val SCHED_DELAY_TU = 10
+  private val ASK_TIMEOUT_TU = 30
```

Comment:
Whoops, these are still really small.

Is 15 mins still the delay we're sticking with?

Reply:
Ask timeout and publish interval could be same. 

We should make the publish interval configurable so it can be changed easily in the future.

Reply:
Done!